src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,114 @@
+            PreparedStatement statement =
+                    connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery(query);
+            if (resultSet.next()) {
+                return Optional.of(retrieveProductFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find product with id " + id
+                    + " in the database.", e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM products";
+            PreparedStatement statement =
+                    connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery(query);
+            List<Product> products = new ArrayList<>();
+            while (resultSet.next()) {
+                products.add(retrieveProductFromResultSet(resultSet));
+            }
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't retrieve products from the database.", e);
+        }
+    }
+
+    @Override
+    public Product update(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE products SET name = ?, price = ? WHERE id = ?";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setBigDecimal(2, BigDecimal.valueOf(item.getPrice()));
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                item.setId(resultSet.getLong("product_id"));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add " + item + " to the database.", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "DELETE FROM products WHERE id = ?";
+            PreparedStatement statement =
+                    connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery(query);
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete product with id " + id
+                    + " from the database.", e);
+        }
+    }
+
+    private Product retrieveProductFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        String name = resultSet.getString("name");
+        double price = resultSet.getBigDecimal("price").doubleValue();
```

Comment:
As far as I remember you can just get double from resultSet via resultSet.getDouble("price");

Reply:
So far as I understand, doubleValue() turns a BigDecimal into a Double. I'm not sure I want to change Double to BigDecimal everywhere in 'price' in my old code, so I tried this)